### PR TITLE
WT-8513 Conditionally eliminate the empty space on FLCS pages (and fix bug this exposed)

### PR DESCRIPTION
**Pull request by sauclovian-wt**
If it looks like writing out and reading back in the empty space will be more expensive than moving the time window data, go ahead and memmove the time windows.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by keithbostic** [comment]
```
[2021/12/08 01:13:05.089] src\reconcile\rec_write.c(1311): error C2220: warning treated as error - no 'object' file generated
[2021/12/08 01:13:05.130] src\reconcile\rec_write.c(1311): warning C4267: '-=': conversion from 'size_t' to 'uint32_t', possible loss of data
```

**Comment by sauclovian-wt** [comment]
Ok, I think that's got it.

**Review by keithbostic — APPROVED** [review]

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -1329,6 +1329,54 @@ __wt_rec_split_grow(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t add_len)
...
+    uint8_t *src, *dst;
+
+    /* Total size of page. */
+    totalsize = WT_PTRDIFF32(r->aux_first_free, r->cur_ptr->image.mem);
+
+    /* Size of the entire primary data area, including headers. */
+    primarysize = WT_PTRDIFF32(r->first_free, r->cur_ptr->image.mem);
+
+    /* Size of the empty space. */
+    emptysize = r->aux_start_offset - primarysize - WT_COL_FIX_AUXHEADER_RESERVATION;
```
Comment (keithbostic) [inline src/reconcile/rec_write.c:1349]: Can you please group the changes to force operation ordering? (This always confuses me, tbh.)

**Review by keithbostic — APPROVED** [review]

File: src/reconcile/rec_write.c (hunk 1 above)
Comment (sauclovian-wt) [reply src/reconcile/rec_write.c]: Done.

**Pull request merged**